docs/resources/backuppolicy.md

```diff
@@ -106,19 +106,39 @@ Optional:
 - `daily` (Number) The number of daily backups to retain
 - `hourly` (Number) The number of hourly backups to retain
 - `monthly` (Number) The number of monthly backups to retain
+- `settings` (Attributes) (see [below for nested schema](#nestedatt--retention--settings))
 - `weekly` (Number) The number of weekly backups to retain
 - `yearly` (Number) The number of yearly backups to retain
 
+<a id="nestedatt--retention--settings"></a>
+### Nested Schema for `retention.settings`
+
+Optional:
+
+- `day_of_week` (String) The day of the week used to promote backup to weekly
+- `month` (String) The month of the year used to promote backup to yearly
+- `promote_latest_to_daily` (Boolean) Whether to promote the latest backup within the day if multiple backups exist for that day
+- `promote_latest_to_hourly` (Boolean) Whether to promote the latest backup within the hour if multiple backups exist for that hour
+- `promote_latest_to_monthly` (Boolean) Whether to promote the latest backup within the month if multiple backups exist for that month
+- `relative_to_last` (Boolean) Whether to apply the backup rotation scheme relative to the last successful backup instead to the current time
+
+
 
 <a id="nestedatt--status"></a>
 ### Nested Schema for `status`
 
 Read-Only:
 
+- `last_missed_backups` (List of Object) The last database backups that were not scheduled by this policy (see [below for nested schema](#nestedatt--status--last_missed_backups))
 - `last_missed_schedule_time` (String) The time that backups were last missed by this policy
 - `last_schedule_time` (String) The time that backups were last taken by this policy
 - `next_schedule_time` (String) The time that backups are next scheduled by this policy
 
+<a id="nestedatt--status--last_missed_backups"></a>
+### Nested Schema for `status.last_missed_backups`
+
+Read-Only:
+
```

Comment:
`last_missed_backups` is missing all of its fields

Reply:
Good catch. It looks like Terraform does not provide a way to specify sub-schemas for object lists.

I made a change to the schema generator that tries to extract some useful information and attach it to the Terraform schema, but there does not seem to be any way to attach the descriptions and other metadata for attributes.

I am actually considering removing the `@TerraformAttribute` annotation from the Java code so that we do not have to deal with this, because it's possible that it does not even deserialize correctly and would generate errors.

Reply:
Maybe I am missing something, but then how is `retention.settings` exposed?

Reply:
`retention.settings` is a different attribute and is an object as opposed to a list containing objects.

`last_missed_backups` is the only attribute that is a list containing objects (as opposed to strings or some other primitive type).

Reply:
Okay, I fixed the nested object schema. It turned out to be pretty straight forward once I realized that I'm supposed to use `ListNestedAttribute`.